contracts/strategies/donation-voting/DonationVotingStrategy.sol

```diff
@@ -0,0 +1,451 @@
+                recipient.recipientStatus = InternalRecipientStatus.Appealed;
+                emit Appealed(recipientId, _data, _sender);
+            } else {
+                recipient.recipientStatus = InternalRecipientStatus.Pending;
+                emit Registered(recipientId, _data, _sender);
+            }
+        }
+    }
+
+    function _allocate(bytes memory _data, address _sender)
+        internal
+        virtual
+        override
+        nonReentrant
+        onlyActiveAllocation
+    {
+        (address recipientId, uint256 amount, address token) = abi.decode(_data, (address, uint256, address));
+
+        Recipient storage recipient = _recipients[recipientId];
+
+        if (!_isTokenAllowed(token) || recipient.recipientStatus != InternalRecipientStatus.Accepted) {
+            revert INVALID();
+        }
+
+        _transferAmount(token, address(this), amount);
+
+        claims[recipientId][token] += amount;
+
+        emit Allocated(recipientId, amount, token, _sender);
+    }
+
+    function _distribute(address[] memory __recipientIds, bytes memory, address _sender)
+        internal
+        virtual
+        override
+        onlyPoolManager(_sender)
+        onlyAfterAllocation
+    {
+        uint256 recipientLength = __recipientIds.length;
+        for (uint256 i; i < recipientLength;) {
+            address recipientId = __recipientIds[i];
+
+            Recipient storage recipient = _recipients[recipientId];
+
+            if (recipient.recipientStatus != InternalRecipientStatus.Accepted) {
+                revert INVALID();
+            }
+
+            uint256 amount = payoutSummaries[recipientId].amount;
+
+            IAllo.Pool memory pool = allo.getPool(poolId);
+            _transferAmount(pool.token, recipient.recipientAddress, amount);
+
+            emit Distributed(recipientId, recipient.recipientAddress, amount, _sender);
+            unchecked {
+                i++;
+            }
+        }
+    }
+
+    /// @notice Check if sender is identity owner or member
+    /// @param _anchor Anchor of the identity
+    /// @param _sender The sender of the transaction
+    function _isIdentityMember(address _anchor, address _sender) internal view returns (bool) {
+        IRegistry registry = allo.getRegistry();
+        IRegistry.Identity memory identity = registry.getIdentityByAnchor(_anchor);
+        return registry.isOwnerOrMemberOfIdentity(identity.id, _sender);
+    }
+
+    /// @notice Get the recipient
+    /// @param _recipientId Id of the recipient
+    function _getRecipient(address _recipientId) internal view returns (Recipient memory) {
+        return _recipients[_recipientId];
+    }
+
+    /// @notice Check if token is allowed for allocation
+    function _isTokenAllowed(address _token) internal view returns (bool) {
+        uint256 allowedTokensLength = allowedTokens.length;
+
+        if (allowedTokensLength == 0) {
+            return true;
+        }
+
```

Comment:
whole function can be removed or should return value of allowedToken mapping

Reply:
removed